tests/installation/upgrade_select.pm

```diff
@@ -42,7 +42,25 @@ sub run {
             send_key $cmd{ok};
         }
         else {
-            assert_screen 'migration_target_' . lc(get_var('SLE_PRODUCT', 'sles')), 120;
+            send_key 'alt-p';
+            # Confirm default migration target matches correct base product
+            my $migration_target_base = 'migration_target_' . lc(get_var('SLE_PRODUCT', 'sles')) . lc(get_var('VERSION'));
+            assert_screen $migration_target_base, 120;
+            # Confirm other migration targets match the same base product
+            # Assume no more than 6 possible migration targets
+            for (1 .. 5) {
+                send_key 'down';
+                unless (check_screen $migration_target_base) {
+                    record_soft_failure 'bsc#1071185: Incorrect migration target';
```

Comment:
bug currently have status "RESOLVED INVALID"

Reply:
Yes, it was marked as "INVALID" because developer thought the feature was not implemented yet. (See https://bugzilla.suse.com/show_bug.cgi?id=1071185#c5)
But it is actual issue during migration at this point.
Should I remove the bug ID and only keep the failure reason in record_soft_failure?

Reply:
Having record soft failure pointing to invalid bug or having soft failure without a bug at all - both options looks wrong ....
Does there fate entry describing that this flow suppose to be implemented ?

Reply:
Yes, the feature has been described in https://fate.suse.com/323163 and https://fate.suse.com/323165, but none of which has been fully implemented yet.
Personally, I don't think the issue is "INVALID", maybe just a little earlier to file the bug.
If a soft_failure with invalid bug is not acceptable, then I will submit this commit again later when fate#323163 and fate#323165 are marked as "Done". At that time, there should NOT be issue bsc#1071185, then I can update test code to remove the workaround.

Reply:
This part is not implemented yet from the fates, so i think this workaround assumes a possible 'correct' behaviour to let the test is not interruptted here. what about to use `record_info('workaround', "we know what we are doing")` to explian here until the fate is done?
wdyt @QingmingSu @asmorodskyi

Reply:
I think a `record_info` should always have a reference to an external ticket at least. In this case I suggest to use one of the fate's as reference. A "soft fail" with no bugref will just cause false alarms and people will slowly accept a "soft fail" as "pass" which it is not. It's a failure with a known ticket reference.

Reply:
what if just replace bug id with fate entry ? say something "waiting for fates #323163 and #323165 to be implemented "?

Reply:
Thanks for all your help, I want to record a failure once an incorrect migration target detected but keep the test going forward without break. So the record_info is really what I need:
record_info 'Likely error detected', 'Incorrect migration target? See https://fate.suse.com/323165', result => 'fail';

Even after #323163 and #323165 being implemented, we can use the same test code to detect invalid migration targets.